tRPC - Case Study

Menubrea opened the pull request:
While there are some things to improve upon, such as spelling errors, sentence structure, reference structure and so on, I think the case study is in a place where I think I could need some valuable feedback on what to improve, potential issues if any or other suggestions you might have.

Fermain commented on trpc-case-study.md:
```diff
@@ -0,0 +1,275 @@
+#### tRPC uses Typescript
+This can be viewed as both a strenght and a weakness. On one hand, it allows for a uniform coding language and infrastructure between backend and frontend, improving worklflow and communcation between the team. 
+
+While on the other hand, if Typescript is not the preferred programming language, or if team structure is heavily silo'd between front- and backend tRPC might not be the best option.
```
```suggestion
While on the other hand, if Typescript is not the preferred programming language, or if team structure is heavily siloed between front and backend tRPC might not be the best option.
```

Fermain commented on trpc-case-study.md:
```diff
@@ -0,0 +1,275 @@
+## Weaknesses
+
+#### tRPC uses Typescript
+While your front-end certainly has knowledge of Javascript, and possibly also well versed in Typescript. The same might not be the case for your backend team, or it might simply not be the preferred techonology. There is a greater variety in programming lanugages typically used for backend than there is for frontend. 
```
```suggestion
While a front-end team certainly has knowledge of Javascript and may be well versed in Typescript, the same might not be the case for a backend team. There is a greater variety in programming languages typically used for backend than for frontend, meaning that Typescript may not always be the most suitable choice.
```

Fermain commented on trpc-case-study.md:
```diff
@@ -0,0 +1,275 @@
+#### tRPC uses Typescript
+While your front-end certainly has knowledge of Javascript, and possibly also well versed in Typescript. The same might not be the case for your backend team, or it might simply not be the preferred techonology. There is a greater variety in programming lanugages typically used for backend than there is for frontend. 
+
+While strict static typing leaves a lot of benefits, it can be frustrating to and cumbersome to learn, especially for junior level developers.
```
```suggestion
While strict static typing leaves a lot of benefits, it can be frustrating and cumbersome to learn, especially for junior-level developers.
```

Fermain commented on trpc-case-study.md:
```diff
@@ -0,0 +1,275 @@
+While strict static typing leaves a lot of benefits, it can be frustrating to and cumbersome to learn, especially for junior level developers.
+
+#### Works best in a monorepo
+Since many of tRPC's features benefit from using a singular repository, it might not be the best option if the development team has a need or requirement for heavily silo'd frontend and backend.
```
```suggestion
Since many of tRPC's features benefit from using a singular repository, it might not be the best option if the development team has a need or requirement for a heavily siloed frontend and backend.
```

Fermain replied:
Contractions are quite rare in English and it's typically safer to go with `[-ed]` unless it looks really strange. Siloed looks pretty weird but is indeed correct.

Fermain commented on trpc-case-study.md:
```diff
@@ -0,0 +1,275 @@
+
+#### Scalability
+- In larger scale projects, in a larger team, its often beneficial to come up with a clear set of rules and a schema that everyone can work from. tRPC does not offer this.
+- If your app is getting increasingly more complex and require more nuanced requirements, tRPC might not be the best option.
```
Why not?

Menubrea replied:
I've resolved this by explaining that other options provide better solutions for increasingly scaling complextion. So while there is nothing wrong with using tRPC in this scenario. The project might benefit more from a different solution.

Fermain commented on trpc-case-study.md:
```diff
@@ -0,0 +1,275 @@
+
+*Use GraphQL if:*
+- Its a large scale project, with the need of well defined schemas, maintenance and future development requirements.
+- if your team is either heavily silo'd or use multiple programming languages.
```
```suggestion
- if your team is either heavily siloed or use multiple programming languages.
```

Fermain commented on trpc-case-study.md:
```diff
@@ -0,0 +1,275 @@
+- if your team is either heavily silo'd or use multiple programming languages.
+
+*Use tRPC if:*
+- You are looking to learn to create your own API's.
```
```suggestion
- You are looking to learn to create your own APIs.
```
In this case `API's` could mean:
- Belonging to API (the API's test methods are passing)
- API is ... (the API's broken again)

But not:
- More than one API (the APIs are working correctly)

Fermain commented on trpc-case-study.md:
```diff
@@ -0,0 +1,275 @@
+
+## Summary
+
+tRPC offers an excellent introduction to creating API's. Using it is intuitive, fast and efficient. With some experience, you can have functional endpoints up and running within minutes. Whether you are a junior when it comes to backend or web-development in general, or an experienced full-stack developer tRPC should offer a lot of welcome features for your next project - increasing your workflow ...
```
```suggestion
tRPC offers an excellent introduction to creating APIs. Using it is intuitive, fast and efficient. With some experience, you can get functional endpoints up and running within minutes. Whether you are a junior when it comes to backend or web development in general, or an experienced full-stack developer tRPC should offer a lot of welcome features for your next project - increasing your workflow, maintaining your code base and overall helping you become an overall more productive developer. 
```

Fermain reviewed (changes requested):
This is a well researched and researching case study about a protocol I am not familiar with. After reading I have a good understanding of this tool thanks to your explanations. Most of my comments are around spelling/grammar/phrasing rather than content.

Please change all instances of double back tick quotes to single back tick quotes. You can do this quickly in VS Code like so:

<img width="520" alt="image" src="https://user-images.githubusercontent.com/10801540/218399770-19c2eb52-dcfd-465d-9fd0-59b8d74c0de3.png">

Menubrea commented:
## Changelog

- Fixed spelling, grammatical and structural errors. 
- Changed from double ticks to single ticks
- Accepted suggested changs
- Made adjustments to text that was previously contradictory or not well explained.

Menubrea commented on trpc-case-study.md:
```diff
@@ -0,0 +1,275 @@
+#### tRPC is lightweight
+Since tRPC is a protocol or method for accessing backend functionality, it allows for better performing applications without any code generation or runtime bloat.
+
+Since tRPC uses Typescript, which is a statically typed language, type definitions are checked during compiling rather than runtime, increasing speed of application.
```
tRPC advertise the product with no build step, but I understand why this part conflicts with the previous section. As I can't properly explain it, I've decide to cut it out from the case-study.

Fermain reviewed (approved):
An excellent quality contribution that shows deep research and critical thinking. Well done.
